chore(typography): update typography to SHINE styles

dancormier opened the pull request:
[SPARK-102](https://stackoverflow.atlassian.net/browse/SPARK-102)

https://deploy-preview-2028--stacks.netlify.app/product/foundation/typography/#sizes

---

This PR update base typography for Stacks Classic. Since our font sizes largely are based on `rem` units, I've done my best to those sizes to match up to pixel values specified in the Figma. They should all be within a 100th of a pixel or so.

I'll update the visual regression images when this is ready to merge.

> [!NOTE]
> We modify some of the larger font sizes viewports below the `sm` breakpoint (see changes [here](https://github.com/StackExchange/Stacks/pull/2028/files#diff-d1878832efc7bac5eb4e78a190e5aecdb06d413fe983488367898ab5728b8806L45-L56) and/or the table in the collapsed section below). I used my best judgement. If you'd like me to adjust, let me know @CGuindon!

## How to test
- Go to the [typography docs](https://deploy-preview-2028--stacks.netlify.app/product/foundation/typography/#sizes)
- Verify that the font sizes shown match the sizes specified in the [Figma](https://www.figma.com/design/do4Ug0Yws8xCfRjHe9cJfZ/Project-SHINE---Product-UI?node-id=2268-22049&t=5ggsDIb9OENEwDl0-1)
- Resize the browser below 640px and checkout those sizes
- (optional) verify the sizes shown match the rendered sizes. You can do this by opening the inspector on a given font value, selecting the "computed" tab, and looking for the value of `font-size`

## TODO before merge
- [ ] Update Less snapshots
- [ ] Update visual regression images

<details>
<summary>Table of font sizes + Screenshots</summary>

## Sizes (as of Thu Oct 30, 2025)

  | Standard | Mobile
-- | -- | --
Fine | 12px | 12px
Caption | 13px | 13px
Body 1 | 14px | 14px
Body 2 | 16px | 15px
Body 3 | 17px | 16px
Subheading | 20px | 18px
Title | 22px | 20px
Headline 1 | 28px | 22px
Headline 2 | 36px | 25px
Display 1 | 46px | 28px
Display 2 | 58px | 32px
Display 3 | 72px | 36px
Display 4 | 100px | 42px

## Screenshots

### Standard sizes
<img alt="image" src="https://github.com/user-attachments/assets/59444a57-84fe-487c-9bac-8c0cad734f11" />

### Mobile sizes (> `sm` breakpoint)

<img alt="image" src="https://github.com/user-attachments/assets/8495b723-25f7-4c0a-adde-d7c2041f9bc0" />

</details>

dancormier committed:
chore(typography): update type sizes; rm .fs-category

dancormier committed:
Merge branch 'beta' into SPARK-102/typography

dancormier committed:
add changeset

dancormier committed:
Update font weight classes

ttaylor-stack commented on packages/stacks-classic/lib/atomic/typography.less:
```diff
@@ -39,34 +39,21 @@ p {
 .fs-body1 { font-size: var(--fs-body1) !important; }
 
 .fs-caption,
```
fs-caption is showing as 12px but in the Figma it shows it should be 13px

<img width="1919" height="1086" alt="image" src="https://github.com/user-attachments/assets/31d35dd2-cf79-474d-b322-eb725c496e89" />

dancormier replied:
Great catch @ttaylor-stack! Fixed in https://github.com/StackExchange/Stacks/pull/2028/commits/35996044b4106450abe31f5b4c006f321bd374e7

ttaylor-stack commented on packages/stacks-docs/_data/typography.json:
```diff
@@ -2,87 +2,81 @@
     },
     {
       "term": "body3",
       "define": "17px",
```
Should this be 18px?

<img width="1919" height="1085" alt="image" src="https://github.com/user-attachments/assets/498d13a1-4ea2-4cbd-86b9-1fa187b6864c" />

dancormier replied:
Ah, yep! It's resolved in https://github.com/StackExchange/Stacks/pull/2028/commits/35996044b4106450abe31f5b4c006f321bd374e7

ttaylor-stack reviewed (changes requested):
Just a few small things

dancormier committed:
Address review

dancormier committed:
Merge branch 'beta' into SPARK-102/typography

ttaylor-stack commented on packages/stacks-classic/lib/atomic/typography.less:
```diff
@@ -89,17 +75,18 @@ p {
-#stacks-internals #responsify('.ta-right', { text-align: right !important; });
+#stacks-internals #responsify('.ta-left', { text-align: left !important; });;;
+#stacks-internals #responsify('.ta-center', { text-align: center !important; });;;
+#stacks-internals #responsify('.ta-right', { text-align: right !important; });;;
```
I think you might have the same issue I had where prettier is adding extra semicolons

ttaylor-stack reviewed (changes requested):
Prettier issue

CGuindon reviewed (changes requested):
JustI would just suggest bumping up the responsive small screen sizes up by one from the old sizing as well. Otherwise LGTM
<img width="483" height="561" alt="Screenshot 2025-10-31 at 11 20 52 AM" src="https://github.com/user-attachments/assets/fb48d12e-0e60-4f7b-9589-4737e015c186" />

dependabot[bot] committed:
build(deps-dev): bump stylelint from 16.24.0 to 16.25.0 (#2032)

Bumps [stylelint](https://github.com/stylelint/stylelint) from 16.24.0 to 16.25.0.
- [Release notes](https://github.com/stylelint/stylelint/releases)
- [Changelog](https://github.com/stylelint/stylelint/blob/main/CHANGELOG.md)
- [Commits](https://github.com/stylelint/stylelint/compare/16.24.0...16.25.0)

---
updated-dependencies:
- dependency-name: stylelint
  dependency-version: 16.25.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dancormier committed:
Merge branch 'develop' into SPARK-102/typography

dancormier committed:
Fix prettier-induced semicolon psychosis

dancormier committed:
Merge branch 'beta' into SPARK-102/typography

dancormier committed:
Update responsive type sizes

dancormier commented:
@ttaylor-stack @CGuindon I've addressed your reviews so this PR is ready for a re-review. Thank you!

dancormier committed:
Update visual regression images

dancormier committed:
Merge branch 'beta' into SPARK-102/typography
